File: TypeScript/ProjectEuler4.ts
class MultipierList {
    private list: number[] = [];

    constructor() {
        for (var i = 100; i < 999; i++) {
            for (var j = 100; j < 999; j++) {
                this.list.push(i * j);
            }
        }
    }

    getList() {
        return this.list;
    }
}


var list = new MultipierList().getList();
console.log(Math.max.apply(null,list.filter(x=>x.toString() === x.toString().split("").reverse().join(""))));
while (true) {
    
}

File: CSharp/ProjectEuler4.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TestCSharp
{
    public class Program
    {

        public static Func<int, IEnumerable<int>, bool> isDevided = (number, numbers) => numbers.All(x => {return  number % x == 0; });
         
        public static IEnumerable<int> ListGen(int start,int stop)
        {
            for (int i = start; i < stop; i++)
            {
                for (int j = start; j < stop; j++)
                {
                    yield return i * j;
                }
            }
        }

        static void Main(string[] args)
        {
            var result = ListGen(100, 900).Where(x => x.ToString() == string.Join("", x.ToString().ToCharArray().Reverse())).Max();
            Console.WriteLine(result);
        }
    }
}
